packages/core/src/interfaces.ts

```diff
@@ -3,26 +3,52 @@ export interface ValidationOptions {
     runSerially?: boolean;
 }
 
-export interface BasicValidation {
-    module: string;
-    config?: any;
-    type: 'single' | 'bulk';
-    transformation?: any;
+export type ConstraintConfiguration = Partial<Record<string,any>> & {
+    __prerequisites?: ConstraintsMapping
 }
 
-export interface Validation extends BasicValidation {
-    description: string;
-    condition?: BasicValidation;
+export type ConstraintValue = boolean | string | number | ConstraintConfiguration;
+
+export interface ConstraintsMapping {
+    [path: string] : {
+        [constraintName: string] : ConstraintValue | [ConstraintValue],
+    } & {
+        __and?: (string | ConstraintsMapping)[]
+        __or?: (string | ConstraintsMapping)[]
+    }
+}
+
+/**
+ * This interface contains relevant references for the execution of the constraints such as the reference to '__self' object and '__parent' runtimecontext
+ */
+export interface IRuntimeContext {
```

Comment:
If we keep the convention of @Ofri-Chen, then this should be named `RuntimeContext`.

his convention is to add `I` to logical interfaces like `IValidator,` and to omit the `I` for data models, like `ValidationError`

Reply:
Understood